java/ql/src/semmle/code/java/frameworks/spring/SpringWebUtil.qll

```diff
@@ -0,0 +1,241 @@
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Element of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Element of Argument[1];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;MapKey of Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;MapKey of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;Element of MapValue of Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;Element of MapValue of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replacePath;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replacePath;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;replacePath;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQuery;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replaceQuery;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQuery;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;(String,Collection);;Element of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;(String,Collection);;Element of Argument[1];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;(String,Object[]);;ArrayElement of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;(String,Object[]);;ArrayElement of Argument[1];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;Element of MapKey of Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;Element of MapValue of Argument[0];ReturnValue;taint",
```

Comment:
Looks like a fluent method? Then the other two lines should target `Argument[-1]` as that will automatically add the steps to `ReturnValue`.
```suggestion
        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;Argument[-1];ReturnValue;value",
        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;Element of MapKey of Argument[0];Argument[-1];taint",
        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;Element of MapValue of Argument[0];Argument[-1];taint",
```